client/v2/influxdb.go

```diff
@@ -0,0 +1,218 @@
+package client
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/influxdb/influxdb/models"
+)
+
+type Config struct {
+	URL       *url.URL
+	Username  string
+	Password  string
+	UserAgent string
+	Timeout   time.Duration
+}
+
+type BatchPointsConfig struct {
+	Precision        string
+	Database         string
+	RetentionPolicy  string
+	WriteConsistency string
+}
+
+type Client interface {
+	Write(pb BatchPoints) error
+}
+
+func NewClient(conf Config) Client {
+	if conf.UserAgent == "" {
+		conf.UserAgent = "InfluxDBClient"
+	}
+	return &client{
+		url:        conf.URL,
+		username:   conf.Username,
+		password:   conf.Password,
+		useragent:  conf.UserAgent,
+		httpClient: &http.Client{Timeout: conf.Timeout},
+	}
+}
+
+type client struct {
+	url        *url.URL
+	username   string
+	password   string
+	useragent  string
+	httpClient *http.Client
+}
+
+type BatchPoints interface {
+	AddPoint(p *Point)
+	Points() []*Point
+
+	Precision() string
+	SetPrecision(s string)
+
+	Database() string
+	SetDatabase(s string)
+
+	WriteConsistency() string
+	SetWriteConsistency(s string)
+
+	RetentionPolicy() string
+	SetRetentionPolicy(s string)
+}
+
+func NewBatchPoints(c BatchPointsConfig) BatchPoints {
+	return &batchpoints{
+		database:         c.Database,
+		precision:        c.Precision,
+		retentionPolicy:  c.RetentionPolicy,
+		writeConsistency: c.WriteConsistency,
+	}
+}
+
+type batchpoints struct {
+	sync.Mutex
+
+	points           []*Point
```

Comment:
yes, I was debating doing that, but after talking to Jason I decided it would be better to have a thin wrapper, which has two benefits:

* We decide which methods are public, so for the client, we can make a point be immutable once it's created
* Users don't need to import and create `models.Point` objects to work with the client